### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string.

### ISSUE THREAD

**Issue by ElRafa0** [TannerGabriel/discord-bot#50]
I took everything from your website and tried it out, but there is something wrong with the code. When i join vc, the bot joins too, but he dont plays the music how it should.

The error: 

TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received undefined
    at validateString (internal/validators.js:124:11)
    at Url.parse (url.js:159:3)
    at Object.urlParse [as parse] (url.js:154:13)
    at Object.exports.getURLVideoID (C:\Users\gomes\Desktop\DiscordBot\node_modules\ytdl-core\lib\url-utils.js:30:22)
    at Object.exports.getVideoID (C:\Users\gomes\Desktop\DiscordBot\node_modules\ytdl-core\lib\url-utils.js:63:20)
    at Function.exports.<computed> [as getInfo] (C:\Users\gomes\Desktop\DiscordBot\node_modules\ytdl-core\lib\info.js:364:23)
    at ytdl (C:\Users\gomes\Desktop\DiscordBot\node_modules\ytdl-core\lib\index.js:19:8)
    at play (C:\Users\gomes\Desktop\DiscordBot\main.js:114:13)
    at execute (C:\Users\gomes\Desktop\DiscordBot\main.js:74:9)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  code: 'ERR_INVALID_ARG_TYPE'
}
(node:16248) UnhandledPromiseRejectionWarning: DiscordAPIError: Cannot send an empty message
    at RequestHandler.execute (C:\Users\gomes\Desktop\DiscordBot\node_modules\discord.js\src\rest\RequestHandler.js:154:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async RequestHandler.push (C:\Users\gomes\Desktop\DiscordBot\node_modules\discord.js\src\rest\RequestHandler.js:39:14)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:16248) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:16248) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by OmarYC** [comment]
I have the same issue

**Comment by TannerGabriel** [comment]
Thank you for mentioning this issue and sorry for the late response.

It seems that the `ytdl-core` library had an update, and the parameter names for the song title and URL changed. I updated the code in the article already and will also update the repo in the next few hours.

Here is what I changed:

From:

```js
const song = {
        title: songInfo.title,
        url: songInfo.video_url,
};
```

To:

```js
const song = {
        title: songInfo.videoDetails.title,
        url: songInfo.videoDetails.video_url,
};
```

Let me know if this fixes your issue.
